### ISSUE TITLE
Reduce BPF related memory allocations

### ISSUE THREAD

**Issue by brancz** [parca-dev/parca-agent#1031]
When trying to deploy parca-agent on low-memory nodes I get various memory related errors:

```
libbpf: Error in bpf_create_map_xattr(unwind_table_1):Operation not permitted(-1). Retrying without BTF.
libbpf: map 'unwind_table_1': failed to create: Operation not permitted(-1)
libbpf: permission error while running as root; try raising 'ulimit -l'? current value: 4096.0 MiB
libbpf: failed to load object 'parca'
level=error name=parca-agent ts=2022-11-16T08:38:09.3139736Z caller=main.go:153 err="load bpf object: failed to load BPF object: operation not permitted"
```

**Comment by javierhonduco** [comment]
Just in case somebody bumps into this, I am working on a fix, need to do some more testing once I have the cycles. Will most likely be ready for review this evening or tomorrow morning 😄

**Comment by brancz** [comment]
Closed by #1042
